### PR TITLE
Drop `installed-state` from installation

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/44814

This removes the `installed-state` IstioOperator. This resource generally just caused confusion for users; they would see it and rely on it to be accurate (its not, since it is static) or change it and expect things to change (they won't, since they are not using in-cluster operator).

There was one command that depended on it, `istioctl x revision`. This command has been removed. Its functionality has been folded into `istioctl tag list`, which can now display revisions without tags.

As an `IstioOperator` is no longer needed, the CRD is removed as well. Note when using the in-cluster operator, the CRD is still used (it is a separate file)

### CODE REVIEW

**Review by bleggett — APPROVED** [review]
Approach LGTM but I think this conflicts with https://github.com/istio/istio/pull/47857

**Comment by hanxiaop** [comment]
> Approach LGTM but I think this conflicts with #47857

I've closed mine, this is a much desired approach.

**Review by hanxiaop — APPROVED** [review]
Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.

**Comment by howardjohn** [comment]
> Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.

I think you can pass it the IOP though?

TBH I would be in favor of removing that too... there isn't much need to verify anything, its declarative -- if you are not sure, you can `istioctl install` again

**Comment by dhawton** [comment]
Do we need a release note to detail the removal of the istioctl x command?

File: istioctl/cmd/root.go [hunk 1]
```diff
@@ -204,7 +203,6 @@ debug and diagnose their Istio mesh.
 	experimentalCmd.AddCommand(wait.Cmd(ctx))
 	experimentalCmd.AddCommand(config.Cmd())
 	experimentalCmd.AddCommand(workload.Cmd(ctx))
-	experimentalCmd.AddCommand(revision.Cmd(ctx))
```
Comment (dhawton) [inline istioctl/cmd/root.go:207]: Do we need a release note for this? It's experimental, so I'm not sure.. I know we typically document its promotion and then removal later.

File: releasenotes/notes/drop-iop.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+releaseNotes:
+  - |
+    **Removed** the experimental `istioctl x revision` command. Revisions can be inspected by the stable `istioctl tag list` command.
```
Comment (dhawton) [inline releasenotes/notes/drop-iop.yaml:6]:
```suggestion
    **Removed** the `istioctl experimental revision` command. Revisions can be inspected by the stable `istioctl tag list` command.
```

**Review by linsun — COMMENTED** [review]
overall lgtm

**Comment by bleggett** [comment]
> > > Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.
> > 
> > 
> > I think you can pass it the IOP though?
> > TBH I would be in favor of removing that too... there isn't much need to verify anything, its declarative -- if you are not sure, you can `istioctl install` again
> 
> I think telling user to run "istioctl install" again isn't a great experience.

If we want to retain the UX of `verify-install` we can do it by running install 2x and diffing the results, I think.

**Comment by linsun** [comment]
> > > > Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.
> > > 
> > > 
> > > I think you can pass it the IOP though?
> > > TBH I would be in favor of removing that too... there isn't much need to verify anything, its declarative -- if you are not sure, you can `istioctl install` again
> > 
> > 
> > I think telling user to run "istioctl install" again isn't a great experience.
> 
> If we want to retain the UX of `verify-install` we can do it by doing a second dry-run install and diffing the results, I think.

interesting - that allows us to not deprecate this cmd as it is graduated. @hanxiaop @howardjohn thoughts?

**Review by ericvn — COMMENTED** [review]
LGTM, but agree with the release note questions raised by others.

**Comment by ericvn** [comment]
> Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.

Just noting that we do have one plan in the current docs with verify-install: https://istio.io/latest/docs/setup/install/istioctl/#verify-a-successful-installation

**Comment by howardjohn** [comment]
> > Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.
> 
> Just noting that we do have one place in the current docs with verify-install: https://istio.io/latest/docs/setup/install/istioctl/#verify-a-successful-installation

That usage still works, only doesn't work without `-f`

**Comment by howardjohn** [comment]
Update release notes

**Comment by hanxiaop** [comment]
> > > > > Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.
> > > > 
> > > > 
> > > > I think you can pass it the IOP though?
> > > > TBH I would be in favor of removing that too... there isn't much need to verify anything, its declarative -- if you are not sure, you can `istioctl install` again
> > > 
> > > 
> > > I think telling user to run "istioctl install" again isn't a great experience.
> > 
> > 
> > If we want to retain the UX of `verify-install` we can do it by doing a second dry-run install and diffing the results, I think.
> 
> interesting - that allows us to not deprecate this cmd as it is graduated. @hanxiaop @howardjohn thoughts?

`verify-install` still works but cannot differentiate between different IOP installations, as there's no IOP now. We don't need to install again since we actually have a `--verify` flag for the install command, which verifies components after installation. This is somewhat dup with the current `verify-install` command. Deprecating it also makes sense since we have the verify flag for the install command, and the `verify-install` command is actually too superficial for effectively verifying installation failures.

**Comment by linsun** [comment]
> > > > > > Current changes LGTM. I think the `verify-install` command may also need an IOP file in some cases.
> > > > > 
> > > > > 
> > > > > I think you can pass it the IOP though?
> > > > > TBH I would be in favor of removing that too... there isn't much need to verify anything, its declarative -- if you are not sure, you can `istioctl install` again
> > > > 
> > > > 
> > > > I think telling user to run "istioctl install" again isn't a great experience.
> > > 
> > > 
> > > If we want to retain the UX of `verify-install` we can do it by doing a second dry-run install and diffing the results, I think.
> > 
> > 
> > interesting - that allows us to not deprecate this cmd as it is graduated. @hanxiaop @howardjohn thoughts?
> 
> `verify-install` still works but cannot differentiate between different IOP installations, as there's no IOP now. We don't need to install again since we actually have a `--verify` flag for the install command, which verifies components after installation. This is somewhat dup with the current `verify-install` command. Deprecating it also makes sense since we have the verify flag for the install command, and the `verify-install` command is actually too superficial for effectively verifying installation failures.

Thanks, I think I'm good. I also asked around, no one seem to be impacted by this change from @howardjohn

**Comment by linsun** [comment]
> Update release notes

Don't see the updated one yet... otherwise lgtm

**Comment by howardjohn** [comment]
> > Update release notes
> 
> Don't see the updated one yet... otherwise lgtm

https://github.com/istio/istio/pull/47860/files#diff-825db724670cf0a65e540b201231f4ea6a591b7c15f777a78bcb81ca4abfc99cR12

**Review by linsun — APPROVED** [review]

**Pull request merged**